tests/transforms/test_get_unitary_matrix.py

```diff
@@ -0,0 +1,420 @@
+        return qml.expval(qml.PauliZ(1))
+
+    get_matrix = get_unitary_matrix(my_quantum_function, wire_order=[1, 0, 4, 2, 3])
+    matrix = get_matrix()
+
+    CNOT10 = np.array([[1, 0, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0], [0, 1, 0, 0]])
+
+    expected_matrix = (
+        reduce(np.kron, [I, I, X, I, I])
+        @ reduce(np.kron, [I, I, I, qml.CRZ(0.2, wires=[2, 3]).matrix])
+        @ reduce(np.kron, [Y, I, I, I, I])
+        @ reduce(np.kron, [CNOT10, I, I, I])
+        @ reduce(np.kron, [I, Z, I, I, I])
+    )
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_invalid_argument():
+    """Assert error raised when input is neither a tape, QNode, nor quantum function"""
+
+    get_matrix = get_unitary_matrix(qml.PauliZ(0))
+
+    with pytest.raises(ValueError, match="Input is not a tape, QNode, or quantum function"):
+        matrix = get_matrix()
+
+
+def test_get_unitary_matrix_wrong_function():
+    """Assert error raised when input function is not a quantum function"""
+
+    def testfunction(x):
+        return x
+
+    get_matrix = get_unitary_matrix(testfunction, [0])
+
+    with pytest.raises(ValueError, match="Function contains no quantum operation"):
+        matrix = get_matrix(1)
+
+
+def test_get_unitary_matrix_interface_tf():
+    """Test with tensorflow interface"""
+
+    tf = pytest.importorskip("tensorflow")
+
+    dev = qml.device("default.qubit", wires=3)
+
+    def circuit(beta, theta):
+        qml.RZ(beta, wires=0)
+        qml.RZ(theta[0], wires=1)
+        qml.CRY(theta[1], wires=[1, 2])
+        return qml.expval(qml.PauliZ(1))
+
+    qnode_tensorflow = qml.QNode(circuit, dev, interface="tf")
+
+    get_matrix = get_unitary_matrix(qnode_tensorflow)
+
+    beta = 0.1
+    theta = tf.Variable([0.2, 0.3])
+
+    matrix = get_matrix(beta, theta)
+
+    theta_np = theta.numpy()
+    matrix1 = np.kron(qml.RZ(beta, wires=0).matrix, np.kron(qml.RZ(theta_np[0], wires=1).matrix, I))
+    matrix2 = np.kron(I, qml.CRY(theta_np[1], wires=[1, 2]).matrix)
+    expected_matrix = matrix2 @ matrix1
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_interface_torch():
+    """Test torch interface"""
+
+    torch = pytest.importorskip("torch", minversion="1.8")
+
+    dev = qml.device("default.qubit", wires=3)
+
+    def circuit(theta):
+        qml.RZ(theta[0], wires=0)
+        qml.RZ(theta[1], wires=1)
+        qml.CRY(theta[2], wires=[1, 2])
+        return qml.expval(qml.PauliZ(1))
+
+    qnode_torch = qml.QNode(circuit, dev, interface="torch")
+
+    get_matrix = get_unitary_matrix(qnode_torch)
+
+    theta = torch.tensor([0.1, 0.2, 0.3])
+
+    matrix = get_matrix(theta)
+
+    matrix1 = np.kron(
+        qml.RZ(theta[0], wires=0).matrix, np.kron(qml.RZ(theta[1], wires=1).matrix, I)
+    )
+    matrix2 = np.kron(I, qml.CRY(theta[2], wires=[1, 2]).matrix)
+    expected_matrix = matrix2 @ matrix1
+
+    assert np.allclose(matrix, expected_matrix)
+
+
```

Comment:
Missing an interface test for JAX;  see, e.g., `test_commute_controlled`, for an example of how to set that up (usually the precision of the datatypes needs to manually be cranked).

Reply:
Ok!